libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/MapRouteLine.java

```diff
@@ -170,16 +177,43 @@
 
     // Waypoint attributes
     int originWaypointIcon = typedArray.getResourceId(
-      R.styleable.NavigationMapRoute_originWaypointIcon, R.drawable.ic_route_origin);
+            R.styleable.NavigationMapRoute_originWaypointIcon, R.drawable.ic_route_origin);
     int destinationWaypointIcon = typedArray.getResourceId(
-      R.styleable.NavigationMapRoute_destinationWaypointIcon, R.drawable.ic_route_destination);
+            R.styleable.NavigationMapRoute_destinationWaypointIcon, R.drawable.ic_route_destination);
     typedArray.recycle();
 
     Drawable originIcon = drawableProvider.retrieveDrawable(originWaypointIcon);
     Drawable destinationIcon = drawableProvider.retrieveDrawable(destinationWaypointIcon);
-    belowLayer = findRouteBelowLayerId(belowLayer, style);
 
-    initializeLayers(style, layerProvider, originIcon, destinationIcon, belowLayer);
+    boolean layerBelowExists = false;
+    if (belowLayer != null && !belowLayer.isEmpty()) {
+      for (Layer layer : style.getLayers()) {
+        if (layer.getId().equals(belowLayer)) {
+          layerBelowExists = true;
+          break;
+        }
+      }
+    }
+
+    String belowLayerCandidate = null;
+    if (!layerBelowExists) {
+      for (int i = style.getLayers().size() - 1; i >= 0; i--) {
+        Layer layer = style.getLayers().get(i);
+        if (layer.getId().equals(LocationComponentConstants.SHADOW_LAYER) ||
+                layer.getId().equals(RouteConstants.LAYER_ABOVE_UPCOMING_MANEUVER_ARROW)) {
+          belowLayerCandidate = layer.getId();
```

Comment:
```suggestion
          belowLayer = layer.getId();
```

Reply:
I made belowLayer `final` to make it uumutable

Reply:
Any benefits to making it immutable? Int this case it looks like we're not using it anywhere else so we could mutate it and remove one additional variable.

Reply:
I thought about Kotlin convertation, but it's to little changes to have affect on convertation. I'll revert to mutable field